mappings/doab.py

```diff
@@ -2,129 +2,179 @@
-                ],
-                '{0}|||{1}'
-            )],
-            'title': ('./datacite:title/text()', '{0}'),
-            'is_part_of': [('./dc:relation/text()', '{0}||series')],
-            'publisher': [('./dc:publisher/text()', '{0}||')],
-            'spatial': ('./oapen:placepublication/text()', '{0}'),
-            'dates': [
-                (
-                    [
-                        './datacite:date/text()',
-                        './datacite:date/@type'
-                    ],
-                    '{0}|{1}'
-                )
-            ],
-            'languages': [('./dc:language/text()', '||{0}')],
-            'extent': ('./oapen:pages/text()', '{0} pages'),
-            'abstract': ('./dc:description/text()', '{0}'),
-            'subjects': [('./datacite:subject/text()', '{0}||')],
-            'has_part': [(
-                './dc:identifier/text()',
-                '1|{0}|doab|text/html|{{"reader": false, "download": false, "catalog": false, "embed": true}}'
-            )],
-            'rights': [
-                (
-                    [
-                        './oaire:licenseCondition/@uri',
-                        './oaire:licenseCondition/text()'
-                    ],
-                    'doab|{0}||{1}|'
-                ),
-                (
-                    [
-                        './datacite:rights/@uri',
-                        './datacite:rights/text()'
-                    ],
-                    'doab|{0}||{1}|'
-                )
-            ]
-        }
+    def map_doab_record(self, doab_record, namespaces) -> Record:
+        identifiers, source_id = self._get_identifers(doab_record, namespaces=namespaces)
 
-    def applyFormatting(self):
-        # Set Identifiers
-        self.record.source = 'doab'
-        self.record.identifiers = self.parseIdentifiers()
+        if identifiers is None or len(identifiers) == 0:
+            return None
+
+        title = doab_record.xpath('./dc:title/text()', namespaces=namespaces) + doab_record.xpath('./datacite:creator/text()', namespaces=namespaces)
+        relations = doab_record.xpath('./dc:relation/text()', namespaces=namespaces)
+        publishers = doab_record.xpath('./dc:publisher/text()', namespaces=namespaces)
+        languages = doab_record.xpath('./dc:language/text()', namespaces=namespaces)
+        extents = doab_record.xpath('./oapen:pages/text()', namespaces=namespaces)
+        
+        return Record(
+            uuid=uuid4(),
+            frbr_status=FRBRStatus.TODO.value,
+            cluster_status=False,
+            source=Source.DOAB.value,
+            source_id=source_id,
+            identifiers= identifiers,
+            authors=self._get_authors(doab_record, namespaces=namespaces),
+            contributors=self._get_contributors(doab_record, namespaces=namespaces),
+            title=title[0] if len(title) > 0 else None,
+            is_part_of=[f'{part}||series' for part in relations],
```

Comment:
You may want to add if part checks in these list comprehensions to ensure we aren't adding strings like 'None||series' or '||series'